Example/Auth/Sample/MainViewController.m

```diff
@@ -2528,69 +2594,75 @@ - (void)commontPhoneVerificationWithVerificationID:(NSString *)verificationID
+- (void)updatePhoneNumberWithString:(NSString *_Nullable)phoneNumber
+                         completion:(void(^)(void))completion{
+  [self showSpinner:^{
+    [[AppManager phoneAuthProvider] verifyPhoneNumber:phoneNumber
+                                           UIDelegate:nil
+                                           completion:^(NSString *_Nullable verificationID,
+                                                         NSError *_Nullable error) {
+     if (error) {
+       [self logFailure:@"failed to send verification code" error:error];
+       [self showMessagePrompt:error.localizedDescription];
+       return;
+     }
+     [self logSuccess:@"Code sent"];
+
+     [self showTextInputPromptWithMessage:@"Verification code:"
+                             keyboardType:UIKeyboardTypeNumberPad
+                          completionBlock:^(BOOL userPressedOK,
+                                            NSString *_Nullable verificationCode) {
+       if (!userPressedOK || !verificationCode.length) {
+         return;
+       }
+       [self showSpinner:^{
+         FIRPhoneAuthCredential *credential =
+         [[AppManager phoneAuthProvider] credentialWithVerificationID:verificationID
+                                                     verificationCode:verificationCode];
+         [[self user] updatePhoneNumberCredential:credential
+                                       completion:^(NSError *_Nullable error) {
+           if (error) {
+             [self logFailure:@"update phone number failed" error:error];
+             [self showMessagePrompt:error.localizedDescription];
+           } else {
+             [self logSuccess:@"update phone number succeeded."];
+             if (completion) completion();
+           }
         }];
       }];
     }];
+    [self hideSpinner:^{
+      [self showTypicalUIForUserUpdateResultsWithTitle:kCreateUserTitle error:error];
+   }];
   }];
+ }];
 }
 
-/** @fn linkPhoneNumber
-    @brief Allows linking a verified phone number to the currently signed user.
+/** @fn updatePhoneNumber
+    @brief Allows adding a verified phone number to the currently signed user via popup prompt.
  */
-- (void)linkPhoneNumber {
-  [self showTextInputPromptWithMessage:@"Phone #:"
+- (void)updatePhoneNumber {
+  [self showTextInputPromptWithMessage:@"Update Phone #:"
                           keyboardType:UIKeyboardTypePhonePad
                        completionBlock:^(BOOL userPressedOK, NSString *_Nullable phoneNumber) {
     if (!userPressedOK || !phoneNumber.length) {
       return;
     }
-    [self showSpinner:^{
+    [self updatePhoneNumberWithString:phoneNumber completion:nil];
+ }];
+}
+
+/** @fn linkPhoneNumberWithString
+    @brief Allows linking a verified phone number to the currently signed user.
+    @param phoneNumber Number pass in string for automation.
+ */
+- (void)linkPhoneNumberWithString:(NSString *_Nullable)phoneNumber
+             completion:(void(^)(void))completion{
```

Comment:
The completion should take an error parameter just like signInWithPhoneNumberRecaptchaWithString.